### ISSUE TITLE
Sample project in the repo does not work [NativeAOT]

### ISSUE THREAD

**Issue by netcorefan1** [dotnet/runtimelab#655]
Basically `GetProcAddress` is unable to find the procedure.
Also, it doesn't compile when the Function type definition is declared as `typedef  int (*myFunc)();`
I must miss something obvious.
Attached is a sample that reproduces the issue.
Thanks
[NativeAOTSample.zip](https://github.com/dotnet/runtimelab/files/5941186/NativeAOTSample.zip)

**Comment by jkotas** [comment]
It looks like your project is trying to load the IL .dll, not the native AOT compiled one.

I have fixed a small typo in the sample. With the type fixed, the C portion of the sample compiles and runs fine for me in x64 VS command prompt:
```
**********************************************************************
** Visual Studio 2019 Developer Command Prompt v16.10.0-pre.1.0
** Copyright (c) 2020 Microsoft Corporation
**********************************************************************
[vcvarsall.bat] Environment initialized for: 'x64'

C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview>cd \runtimelab\samples\NativeLibrary

C:\runtimelab\samples\NativeLibrary>cl LoadLibrary.c
Microsoft (R) C/C++ Optimizing Compiler Version 19.28.29828 for x64
Copyright (C) Microsoft Corporation.  All rights reserved.

LoadLibrary.c
Microsoft (R) Incremental Linker Version 14.28.29828.0
Copyright (C) Microsoft Corporation.  All rights reserved.

/out:LoadLibrary.exe
LoadLibrary.obj

C:\runtimelab\samples\NativeLibrary>LoadLibrary.exe
The sum is 10
The concatenated string is okko
```

**Comment by netcorefan1** [comment]
In my specific sample the typo was not present because I removed any defines different than _WIN32, but I am glad that it helped to identify the mistake in the sample. It appears to be that there is another typo in the LoadLibrary.c repo:
`#define PathToLibrary "bin\\Debug\\net5.0\\win-x64\\native\\NativeLibrary.dll"`
should be `#define PathToLibrary "bin\\Debug\\net5.0\\win-x64\\NativeLibrary.dll"` because there is no Native folder generated on compilation.

Unfortunately the problem remains the same. I have included a new sample, this time is the one of the repo and without a c++ console application or any other different things (only change I have done is the path correction).

To reproduce:
`dotnet build Solution.sln`
`cl LoadLibrary.c /Debug /Zi /Od`

Start debugging and you should get this:
<img width="723" alt="2021-02-08-19-34-14" src="https://user-images.githubusercontent.com/63934725/107265721-e78c8b00-6a44-11eb-8306-4fa312f142cf.png">
Running LoadLibrary.exe from x64 Native tools it just return after a couple of seconds from the exception without any message.
Basically, the problem is that `symLoad` does not find the exported procedure and returns a null handle, as I already stated.

[NativeAOT.zip](https://github.com/dotnet/runtimelab/files/5946003/NativeAOT.zip)

**Comment by jkotas** [comment]
> should be #define PathToLibrary "bin\\Debug\\net5.0\\win-x64\\NativeLibrary.dll" because there is no Native folder generated on compilation.

It means that you are not building NativeAOT compiled binary. The NativeAOT is setup to only kick in during `publish`. You may need to change your default build action to publish or setup a custom build step to make it work.

**Comment by netcorefan1** [comment]
Thanks for your tips. I am going near to the solution. This time it generate the native folder, but it only contains native\NativeLibrary.lib, therefore `access(PathToLibrary, F_OK)` fails.
```
C:\Users\max\source\repos\NativeAOT\NativeLibrary>dotnet publish -c Debug -r win-x64 /p:NativeLib=Static /p:SelfContained=true
Microsoft (R) Build Engine version 16.8.3+39993bd9d for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  Restored C:\Users\max\source\repos\NativeAOT\NativeLibrary\NativeLibrary.csproj (in 1.16 sec).
  NativeLibrary -> C:\Users\max\source\repos\NativeAOT\NativeLibrary\bin\Debug\net5.0\win-x64\NativeLibrary.dll
  Generating compatible native code. To optimize for size or speed, visit https://aka.ms/OptimizeCoreRT
  Microsoft (R) Library Manager Version 14.28.29336.0
  Copyright (C) Microsoft Corporation.  All rights reserved.

  "/OUT:bin\Debug\net5.0\win-x64\native\NativeLibrary.lib"
  "obj\Debug\net5.0\win-x64\native\NativeLibrary.obj"
  NativeLibrary -> C:\Users\max\source\repos\NativeAOT\NativeLibrary\bin\Debug\net5.0\win-x64\publish\
```
```
C:\Users\max\source\repos\NativeAOT\NativeLibrary>cl LoadLibrary.c /Debug /Zi /Od
Microsoft (R) C/C++ Optimizing Compiler Version 19.28.29336 for x64
Copyright (C) Microsoft Corporation.  All rights reserved.

LoadLibrary.c
Microsoft (R) Incremental Linker Version 14.28.29336.0
Copyright (C) Microsoft Corporation.  All rights reserved.

/out:LoadLibrary.exe
/debug
LoadLibrary.obj
```
This is what it generates:
```
NativeLibrary\bin\Debug\net5.0\win-x64\native\NativeLibrary.lib
NativeLibrary\bin\Debug\net5.0\win-x64\publish\NativeLibrary.deps.json
NativeLibrary\bin\Debug\net5.0\win-x64\publish\NativeLibrary.lib
NativeLibrary\bin\Debug\net5.0\win-x64\publish\NativeLibrary.pdb
NativeLibrary\bin\Debug\net5.0\win-x64\ref\NativeLibrary.dll
NativeLibrary\bin\Debug\net5.0\win-x64\NativeLibrary.deps.json
NativeLibrary\bin\Debug\net5.0\win-x64\NativeLibrary.dll
NativeLibrary\bin\Debug\net5.0\win-x64\NativeLibrary.pdb
NativeLibrary\LoadLibrary.exe
NativeLibrary\LoadLibrary.ilk
NativeLibrary\LoadLibrary.obj
NativeLibrary\LoadLibrary.pdb
NativeLibrary\vc140.pdb
```
This is time I removed everything. The workspace is just like the repo:
```
Directory.Build.props
Directory.Build.targets
NativeLibrary\Class1.cs
NativeLibrary\LoadLibrary.c
NativeLibrary\NativeLibrary.csproj
NativeLibrary\nuget.config
```
I attached the sample project so that you can reproduce the above steps and see if I have done something wrong, but the files are the same of the repo with no change from my side.
[NativeAOT.zip](https://github.com/dotnet/runtimelab/files/5947630/NativeAOT.zip)
Thanks

**Comment by jkotas** [comment]
> This time it generate the native folder, but it only contains native\NativeLibrary.lib

`/p:NativeLib=Static` produces .lib. If you would like to get .dll, you need to use `/p:NativeLib=Shared`

**Comment by netcorefan1** [comment]
Thanks friend! It works! Now I know everything there is to know and I can finally start to play the games!!
Many thanks, without your help I really don't have any idea if would have ever been able to find a way!
Thanks
